app/routes/api.js

```diff
@@ -39,6 +39,47 @@ router.get('/fact', (req, res) => {
     }
 });
 
+// API endpoint to add a new fact to the database.
+router.post('/fact', (req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+    const { userId, content, discovery_date, note } = req.body;
+
+    // v no longer fails foreign key constraint
+    const submitter_id = userId || 'zzz3737';
+    //const submitter_id = null;
+
+    if (!content) {
+        res.status(400).json({ error: 'Content field is required' });
+        return;
+    }
+
+    //const success = addFact({ submitter_id, content, note, discovery_date });
+    const success = addFact({ submitter_id, content, discovery_date, note });
+
+    if (success) {
```

Comment:
Doesn't addFact(...) return an object with the success property? So you would be looking at success.success to get the boolean value whereas now you're just checking is success is defined or undefined. Let me know if I'm wrong about this.

Reply:
updateFact returns an object, addFact just returns a boolean